### PR TITLE
Add field to identify edge hosts

### PR DESCRIPTION
**Pull request by tahmidefaz**
Subsequent HBI PR: https://github.com/RedHatInsights/insights-host-inventory/pull/892

### CODE REVIEW

File: tests/utils/invalids.py [hunk 1]
```diff
@@ -71,4 +71,5 @@
     {"yum_repos": [{"gpgcheck": "True"}]},
     {"yum_repos": [{"enabled": "False"}]},
     {"installed_packages_delta": ["x" * 513]},
+    {"is_edge": "True"},
```
Comment (thearifismail) [inline tests/utils/invalids.py:74]: valid and invalid both have the same value `True`, how?  Shouldn't `invalid` be some non-boolean value?

Comment (tahmidefaz) [reply tests/utils/invalids.py]: resolving this as we switched to enum. @thearifismail the one in `valid` was a true `boolean` and the one in `invalid` was a `string`

**Comment by tahmidefaz** [comment]
Moving this conversation over from https://github.com/RedHatInsights/insights-host-inventory/pull/892

> @tahmidefaz "is_edge" seems too specific as primary field. Should we make this into more an enum of types (perhaps "host_type")? that would be me extensible i think.

@lphiri Switched to an enum as per our discussion. I only have `edge` as a possible `host_type` as I am not sure what other possible `host_type` options we will have.

If anyone have any more enum suggestions, lmk!

**Comment by lphiri** [comment]
@tahmidefaz LGTM. We don’t have to  define the other typed right away.

**Review by lphiri — APPROVED** [review]

**Review by kruai — APPROVED** [review]
LGTM

**Review by ryandillinfelton — APPROVED** [review]

File: schemas/system_profile/v1.yaml [hunk 2]
```diff
@@ -350,6 +350,12 @@ $defs:
       is_marketplace:
         description: Indicates whether the host is part of a marketplace install from AWS, Azure, etc.
         type: boolean
+      host_type:
+        type: string
+        enum: [edge]
+        maxLength: 4
+        description: Indicates the type of host.
```
Comment (gmcculloug) [inline schemas/system_profile/v1.yaml:357]: This PR was merged while I was typing but still want to ask this question. 🙂   

Should the `description` indicate that this value is derived from other system properties, and not a value explicitly set on the host?  I think most properties here are explicit on the host which makes this one a little different.  @csams thoughts?

**Comment by csams** [comment]
@tahmidefaz @kruai We need one more host_type in the enum to signify a type for existing systems.

**Pull request merged**